staging/src/k8s.io/apiserver/pkg/admission/attributes.go

```diff
@@ -44,21 +44,24 @@ type attributesRecord struct {
 	// But ValidatingAdmissionWebhook add annotations concurrently.
 	annotations     map[string]string
 	annotationsLock sync.RWMutex
+
+	reinvocationState *reinvocationState
```

Comment:
why a pointer?

Reply:
reinvocationState holds mutable state, so the pointer avoid any accidental pass by value.